_posts/2021-05-14-Spring-Boot-TestConfiguration.md

```diff
@@ -0,0 +1,223 @@
+---
+title: Spring Boot @TestConfiguration Annotation
+categories: [spring-boot]
+date: 2021-05-15 00:00:00 +0530
+modified: 2021-05-15 06:00:00 +1000
+author: naveen
+excerpt: "We will see the use of `TestConfiguration` annotation for creating custom beans or overriding specific beans for providing specialized behavior to the application during unit testing of Spring Boot applications ."
+image:
+ auto: 0035-switchboard
+---
+A unit test is used to verify the smallest part of an application ("units") independent of other parts. This makes the verification process easy and fast since the scope of the testing is narrowed down to a class or method. 
+
+`TestConfiguration` annotation is a useful aid for writing unit tests of Spring Boot applications. It allows us to define additional beans or override existing beans in the Spring application context to add specialized configurations for testing. 
+
+In this article, we will see the use of the `TestConfiguration` annotation for writing tests for unit testing of Spring Boot applications.
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/spring-boot/spring-boot-testconfiguration" %}
+
+## Introducing TestConfiguration Annotation
+
+We use the [TestConfiguration](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/context/TestConfiguration.html) to create configurations for our tests. We use this capability for applying customizations for abstracting the external dependencies and run our test in a standalone environment. 
```

Comment:
I don't understand the sentence "We use this capability for applying customizations for abstracting the external dependencies and run our test in a standalone environment". Too abstract, too many complicated words. Can we simplify it?

Reply:
@thombergs I have rewritten this paragraph. Please revisit this section once again, please.